src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,30 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.InvalidRegistrationDataException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new InvalidRegistrationDataException("Register argument is null");
+        }
+        if (user.getLogin() == null || user.getLogin().equals("")) {
+            throw new InvalidRegistrationDataException("Invalid login");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new InvalidRegistrationDataException("User already registered");
+        }
+        if (user.getAge() == null || user.getAge() < 18) {
+            throw new InvalidRegistrationDataException("Invalid age");
+        }
+        if (user.getPassword() == null || user.getPassword().length() < 6) {
+            throw new InvalidRegistrationDataException("Invalid password");
```

Comment:
bad messages inside exception. The user should understand the problem. Why an invalid password? It is null, short length, or something else. Fix it in all messages.